### DISCUSSION TITLE
Should Microsoft commit to building a production application using MAUI?

### DISCUSSION THREAD

**Discussion by RobTF** [dotnet/maui#15109] [Polls]
Microsoft are pushing MAUI as the de-facto app development platform for .NET and the only Microsoft backed solution.

However, I've yet to see one of their own apps use Xamarin or MAUI.

I believe it would be an extremely healthy exercise for Microsoft to commit to building a production, Microsoft branded application using MAUI - released on the appropriate app stores. Not necessarily anything as large as the Office apps or anything, but maybe start with something smaller, like the Azure management app? Something which will turn Microsoft into a consumer as well as a producer of the MAUI SDK.

**Comment by waynemunro** [comment]
Does the term DogFood sound familiar? Not sure if it permitted anymore?

**Comment by saint4eva** [comment]
Maui is not the only Microsoft backed solution - Microsoft supports react native as well.

And some other technologies Microsoft are using, have been in use before Maui came out. DO you expect them to rewrite working apps just to dogfood Maui?

Notwithstanding, new software can easily use Maui if it meets their needs.

**Reply by waynemunro** [reply]
No, but I certainly expect a tech company to promote their own products over others and show some commitment and innovation.

**Reply by RobTF** [reply]
> Maui is not the only Microsoft backed solution - Microsoft supports react native as well.
> 
> And some other technologies Microsoft are using, have been in use before Maui came out. DO you expect them to rewrite working apps just to dogfood Maui?
> 
> Notwithstanding, new software can easily use Maui if it meets their needs.

Isn't React Native a Meta project? What do you mean by Microsoft supports it? Are they a large contributor to the project or something? or is it somehow tied to another Microsoft backed development tool as a first class integration? Asking out of honest interest.

The poll isn't specific enough as to whether the app should be a rewrite of a current app or a new app... just _something_ to show solidarity with the developers they are keen to sell the MAUI vision to and help them assess how mature the product really is, and whether the current development strategy is working.

**Reply by waynemunro** [reply]
This is a typical Microsoft strategy that tends to backfire. To first let others, innovate then try play catchup, but often it's too late. Innovation is expansive and no tech company can afford not to develop and innovate otherwise they just followers.

**Reply by saint4eva** [reply]
> No, but I certainly expect a tech company to promote their own products over others and show some commitment and innovation.

Microsoft Office 365/ Desktop Office has heavily invested in React Native for Desktop and Web before Maui came out. I do not see Office team abandoning that. 

But somethings such as Azure mobile app and 365 admin app etc that are already Xamarin apps, can be migrated to Maui. And new opportunities can use .NET Maui

**Reply by saint4eva** [reply]
> > Maui is not the only Microsoft backed solution - Microsoft supports react native as well.
> > And some other technologies Microsoft are using, have been in use before Maui came out. DO you expect them to rewrite working apps just to dogfood Maui?
> > Notwithstanding, new software can easily use Maui if it meets their needs.
> 
> Isn't React Native a Meta project? What do you mean by Microsoft supports it? Are they a large contributor to the project or something? or is it somehow tied to another Microsoft backed development tool as a first class integration? Asking out of honest interest.
> 
> The poll isn't specific enough as to whether the app should be a rewrite of a current app or a new app... just _something_ to show solidarity with the developers they are keen to sell the MAUI vision to and help them assess how mature the product really is, and whether the current development strategy is working.

Yes, Microsoft is a large contributor, and have used it in many aspect of their large software solutions. For backends, Microsoft groups usually use .NET - most of them are now running on .NET 6 or .NET 7. 

React/ React Native have good story for the web. Since emergence of Blazor, some groups are now using it for their solution. And the more Blazor mature, the mix of Blazor/ Maui can serve many groups within Microsoft.

**Comment by davidortinau** [comment]
@RobTF [this Microsoft 365 Admin app](https://www.microsoft.com/en-us/microsoft-365/business/manage-office-365-admin-app) is Xamarin.Forms and the team is beta testing their .NET MAUI version now having completed the upgrade process. 

The [Azure mobile app](https://azure.microsoft.com/en-us/get-started/azure-portal/mobile-app) is also a Xamarin app upgrading to .NET that will be releasing sometime this year I expect.

And there are a handful of Dynamics 365 apps using our stuff, and others. We also use the product ourselves to build apps for a variety of internal uses, and with customers as well.

**Reply by RobTF** [reply]
@davidortinau That's really good to know! Thanks for the information - I'm not sure it is well known in the community that these tools are being used by Microsoft internally but it's really reassuring to see these apps out there using Xamarin and hopefully MAUI soon.

After I saw your post I did decompile the Azure app for a look out of interest and saw it uses stuff like AppCenter which isn't supported on MAUI (yet?) which may be hindering their progress?

Perhaps you guys should make more of this somehow? I think it really adds weight to what you're doing and it would be useful to know if internally there are any war stories/workarounds for common issues the community are seeing with respect to problems with `CollectionView`, image loading, memory management etc. I imagine it's possible your internal customers have a unique insight those of us on the outside perhaps don't have.

**Reply by saint4eva** [reply]
@RobTF I believe Maui team has some good announcements to make during BUILD conference.

**Reply by FM1973** [reply]
You can use AppCenter with Maui, but only for crashes and tracking. I´m using it for an app in beta state and it works as expected.

**Comment by Xyncgas** [comment]
Microsoft needs the creativity from you, you're supposed to become the chosen one and make something good, so Microsoft can tell other people to come use MAUI and get them to deploy it via visual studio and Opt-in Azure for quality service

**Reply by waynemunro** [reply]
What's so attractive about other tech from Microsoft perspective? If it's better, why can't they just commit to prioritizing the development and support of their own tech, so it improves. Where is the benefit to contributing to the competitor's tech? I don't get it.

**Comment by RobTF** [comment]
@davidortinau Regarding the projects you mentioned earlier in this thread; would you be able to update us with how are they getting on with moving to MAUI? Are they close to production? It would be interesting to see how MS teams are fairing alongside external development houses.

thanks!

**Comment by akhanalcs** [comment]
MSFT should dog-food their products.
That's the only way to greatly improve a product. Be a leader, not a follower of other's tech like React, React Native etc.

**Reply by saint4eva** [reply]
And Microsoft has been doing that.
